plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py

```
@@ -0,0 +1,166 @@
+from omegaconf.errors import ValidationError
+from redis import Redis
+from rq import Queue
+
+from .rq_launcher import RQLauncher
+
+log = logging.getLogger(__name__)
+
+
+def execute_job(
+    idx: int,
+    overrides: Sequence[str],
+    config_loader: ConfigLoader,
+    config: DictConfig,
+    task_function: TaskFunction,
+    singleton_state: Dict[Any, Any],
+    redis_job_id: str,
+) -> JobReturn:
+    """Calls `run_job`
+    """
+    setup_globals()
+    Singleton.set_state(singleton_state)
+
+    sweep_config = config_loader.load_sweep_config(config, list(overrides))
+    with open_dict(sweep_config):
+        sweep_config.hydra.job.id = redis_job_id
+        sweep_config.hydra.job.num = idx
+    HydraConfig.instance().set_config(sweep_config)
+
+    ret = run_job(
+        config=sweep_config,
+        task_function=task_function,
+        job_dir_key="hydra.sweep.dir",
+        job_subdir_key="hydra.sweep.subdir",
+    )
+
+    return ret
+
+
+def launch(
+    launcher: RQLauncher, job_overrides: Sequence[Sequence[str]], initial_job_idx: int,
+) -> JobReturn:
+    """
+    :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+    :param initial_job_idx: Initial job idx in batch.
+    :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+    """
+    setup_globals()
+    assert launcher.config is not None
+    assert launcher.config_loader is not None
+    assert launcher.task_function is not None
+
+    configure_log(launcher.config.hydra.hydra_logging, launcher.config.hydra.verbose)
+    sweep_dir = Path(str(launcher.config.hydra.sweep.dir))
+    sweep_dir.mkdir(parents=True, exist_ok=True)
+
+    # RQ configuration
+    rq_cfg = launcher.rq
+
+    # Redis configuration
+    redis_cfg = launcher.rq["redis"]
+    is_async = True
+    try:
+        is_async = not redis_cfg["mock"]
+    except ValidationError:
+        pass
```

what are you hiding here?
swallowing exceptions can lead to upset stomach.